main.go

```diff
@@ -320,6 +325,21 @@ func (p Proxy) getNewToken(_ http.ResponseWriter, r *http.Request) error {
 	return nil
 }
 
+// isTrustedBot compares the user agent in the request against a list of trusted bots in the configuration and
+// returns true if the user agent contains one of the configured keywords.
+func (p Proxy) isTrustedBot(r *http.Request) bool {
+	userAgent := strings.ToLower(r.Header.Get("User-Agent"))
+	if userAgent == "" {
+		return false
+	}
+	for _, s := range p.TrustedBots {
+		if strings.Contains(userAgent, s) {
```

Comment:
Do you need to set these to lowercase, or did is `newProxy` invoked and it's already done?

Reply:
Good question. Did it in `newProxy` so it didn't have to be done on every request.